### PR TITLE
GCInfoDecoder cleanup

### PR DESCRIPTION
**Pull request by jkotas**
Remove stale ifdefs and a few portability tweaks to make the decoder easier to reuse in CoreRT

### CODE REVIEW

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by jkotas** [comment]
@janvorli @swaroop-sridhar PTLA

File: src/gcinfo/gcinfoencoder.cpp [hunk 1]
```diff
@@ -1354,83 +1339,6 @@ void GcInfoEncoder::Build()
 #endif
     }
 
-
-#ifdef PARTIALLY_INTERRUPTIBLE_GC_SUPPORTED
-    ///////////////////////////////////////////////////////////////////////
-    // Gather EH information
-    ///////////////////////////////////////////////////////////////////////
-   
-    couldBeLive.ClearAll();
```
Comment (swaroop-sridhar) [inline src/gcinfo/gcinfoencoder.cpp]: Is it intentional to remove  `couldBeLive.ClearAll()` too?

**Comment by jkotas** [comment]
Thanks for catching my mistake - I have added it back.

**Pull request merged**